test.disabled/ejabberd_tests/tests/dynamic_modules.erl

```diff
@@ -10,25 +10,29 @@ save_modules(Domain, Config) ->
 
 ensure_modules(Domain, RequiredModules) ->
     CurrentModules = get_current_modules(Domain),
-    [ensure_module(Domain, Mod, Opts, proplists:get_value(Mod, CurrentModules, stopped))
-        || {Mod, Opts} <- RequiredModules].
+    {ToReplace, ReplaceWith} = to_replace(RequiredModules, CurrentModules, [], []),
+    ok = escalus_ejabberd:rpc(gen_mod_deps, replace_modules,
+                              [Domain, ToReplace, ReplaceWith]).
+
+to_replace([], _CurrentModules, ReplaceAcc, ReplaceWithAcc) ->
+    {lists:usort(ReplaceAcc), ReplaceWithAcc};
+to_replace([RequiredModule | Rest], CurrentModules, ReplaceAcc, ReplaceWithAcc) ->
+    {Mod, Opts} = RequiredModule,
+    {NewReplaceAcc, NewReplaceWithAcc} =
+        case lists:keyfind(Mod, 1, CurrentModules) of
+            false when Opts =:= stopped -> {ReplaceAcc, ReplaceWithAcc};
```

Comment:
Will this ever be executed?

Reply:
It definitely happened in practice. It's enough to have `ensure_modules(Domain, [{mymodule, stopped}])` when `mymodule` is actually started only in one preset.

Reply:
OK